### PR TITLE
Add basic support for Yarn 2 (berry)

### LINKED ISSUE #2030: Add support for Yarn v2
The lockfile that yarn v2 generates is different from the lockfile that yarn v1 generates. Dependabot currently always creates Yarn v1 lockfiles, but yarn v2 can't run `yarn install` using a v1 yarn lockfile. This is the error message received:

```
 YAMLException: bad indentation of a mapping entry at line 4, column 15:
       dependencies:
                   ^
       dependencies:
                   ^
```

### PR DESCRIPTION
**Pull request by ylemkimon**
Fixes #2030.

### CODE REVIEW

File: npm_and_yarn/helpers/test/yarn/fixtures/updater/updated-berry/yarn.lock [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  version: 3.1.0
+  resolution: "is-positive@npm:3.1.0"
+  checksum: 6086c9c824e3fe1592bb838c02be6a70edeecc7b639ff762e0a10469aa50d4214e091140903aef6c6d17c75f31b594a4d061972deacf8432b67b11565e26ff98
+  languageName: node
+  linkType: hard
+
+"left-pad@npm:^1.1.3":
+  version: 1.1.3
+  resolution: "left-pad@npm:1.1.3"
+  checksum: 484abc1a07ba933a72498af329c29d1ebd82f54cd60a4b2e4d2603e258fb752170750121aa7cb897a17d85eac50e5c4f248723c7cf327d2da0118e82e96e44b7
```
Comment (ylemkimon) [inline npm_and_yarn/helpers/test/yarn/fixtures/updater/updated-berry/yarn.lock:18]: For Yarn to update the checksum, it seems it needs fetch the package.

File: npm_and_yarn/helpers/test/yarn/fixtures/updater/updated-berry/yarn.lock [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  languageName: node
+  linkType: hard
+
+"root-workspace-0b6124@workspace:.":
```
Comment (ylemkimon) [inline npm_and_yarn/helpers/test/yarn/fixtures/updater/updated-berry/yarn.lock:22]: Yarn 2 has root workspace.

**Comment by Milo123459** [comment]
@ylemkimon any updates?

**Comment by gcrow** [comment]
Also looking to see this.

**Comment by ylemkimon** [comment]
This PR was just a proof-of-concept to see how feasible it is and get the attention of the Dependabot team. I'll close it to reduce any confusion.

**Pull request closed without merging**